awx/main/tasks.py

```diff
@@ -2632,7 +2675,7 @@ def pre_run_hook(self, inventory_update, private_data_dir):
                 _eager_fields=dict(
                     launch_type="sync",
                     job_type='run',
-                    job_tags='update_{},install_collections'.format(source_project.scm_type),  # roles are never valid for inventory
+                    job_tags='update_{},install_roles,install_collections'.format(source_project.scm_type),  # roles are never valid for inventory
```

Comment:
I'm still on the fence about the choices made here.

Background on change - here we are caching _content_, roles & collections, and marking that cache with an identifier. SCM inventory cannot use roles, only collections. But... an SCM inventory could trigger a local cache update, which later gets used by a job. So yes, we do need to update roles here, just to keep the cache mechanics simple.

However, I'm still not detecting whether the cache exists or not. That would be getting into what #5429 was trying to do. This means that the cache would be blown away and both roles & collections re-downloaded for every SCM inventory update. This is gory because right now it would leave a warning log:

https://github.com/ansible/awx/pull/7643/files#diff-9d4ea1dd908b35fb92eaede4bd10bb46R2409

So options are:
1. downgrade the `warn` log to `debug` or `info`, accept that we're not solving the caching problem for SCM inv
2. check if cache exists here, like in the `RunJob` pre_run_hook

This is a pained decision for me right now. I don't want to creep scope further, but I want this to make sense.


Since the SCM inventory obliged to download _roles_, I'm leaning toward option (2), because it would be seen as a regression for it to start doing 20 GB downloads in all these cases where the project does not even have collection requirements.